File: go/fixture/go-mini/internal/env/env.go
// Package env loads process configuration from environment variables.
package env

import (
	"os"
	"strconv"
)

// Configuration is the configuration.
type Configuration struct {
	NumWorkers    int
	BatchSize     int
	FlapWindowSec int
	QueueAddr     string
	Region        string
}

var Config Configuration //nolint:gochecknoglobals // loaded in main, read everywhere

var region string //nolint:gochecknoglobals // TODO

func init() { //nolint:gochecknoinits // TODO
	region = os.Getenv("REGION")
}

// Load loads the configuration.
func Load() {
	Config = Configuration{
		NumWorkers:    intEnv("NUM_WORKERS", 4),
		BatchSize:     intEnv("BATCH", 64),
		FlapWindowSec: intEnv("FLAP_WINDOW_SEC", 30),
		QueueAddr:     os.Getenv("QUEUE_ADDR"),
		Region:        regionOrDefault(),
	}
}

func regionOrDefault() string {
	if region == "" {
		return "us"
	}
	return region
}

func intEnv(key string, def int) int {
	v := os.Getenv(key)
	if v == "" {
		return def
	}
	n, err := strconv.Atoi(v)
	if err != nil {
		return def
	}
	return n
}

File: py/fixture/py-mini/internal/env/env.py
"""Loads process configuration from environment variables."""

import os
from dataclasses import dataclass


@dataclass
class Configuration:
    """Configuration is the configuration."""

    num_workers: int = 4
    batch_size: int = 64
    flap_window_sec: int = 30
    queue_addr: str = ""
    region: str = "us"


CONFIG = Configuration()

_region = os.environ.get("REGION")


def load() -> None:
    """Load the configuration."""
    global CONFIG  # noqa: PLW0603  # loaded in main, read everywhere
    CONFIG = Configuration(
        num_workers=_int_env("NUM_WORKERS", 4),
        batch_size=_int_env("BATCH", 64),
        flap_window_sec=_int_env("FLAP_WINDOW_SEC", 30),
        queue_addr=os.environ.get("QUEUE_ADDR", ""),
        region=_region_or_default(),
    )


def _region_or_default() -> str:
    if not _region:
        return "us"
    return _region


def _int_env(key: str, default: int) -> int:
    v = os.environ.get(key)
    if not v:
        return default
    try:
        return int(v)
    except ValueError:
        return default
